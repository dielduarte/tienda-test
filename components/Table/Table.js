import React from 'react';
import {
  Table as StyledTable,
  TableGroup,
  HeaderColumn,
  BodyColumn,
  TableContent,
  TableImg
} from './Ui';

function renderHeader(header, key) {
  return <HeaderColumn key={key}>{header}</HeaderColumn>;
}

function Table({ headers, children }) {
  return (
    <StyledTable>
      <thead>
        <TableGroup>
          {headers.map((header, key) => renderHeader(header, key))}
        </TableGroup>
      </thead>
      <tbody>{children}</tbody>
    </StyledTable>
  );
}

export { TableGroup, BodyColumn, TableContent, TableImg };
export default Table;
