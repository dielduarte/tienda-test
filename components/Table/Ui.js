import styled from 'styled-components';

export const Table = styled.table`
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;

  tr:nth-child(odd) {
    background: ${props => props.theme.white};
  }

  tr:nth-child(even) {
    background: ${props => props.theme.lightGray};
  }
`;

export const TableGroup = styled.tr`
  height: 60px;
  border-bottom: 2px solid ${props => props.theme.gray};
  padding: 0 10px;
`;

export const HeaderColumn = styled.th`
  font-size: 12px;
  text-align: left;
`;

export const BodyColumn = styled.td`
  font-size: 14px;
`;

export const TableContent = styled.div`
  background: ${props => props.theme.gray};
  padding: 10px;
  display: inline-block;
`;

export const TableImg = styled.img`
  width: 15px;
  height: 15px;
  object-fit: cover;
`;
