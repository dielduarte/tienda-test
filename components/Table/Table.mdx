---
name: Table
---

import { Playground } from 'docz';
import Table, { TableGroup, BodyColumn } from './Table';

# Table

<Playground>
  <Table 
    headers={['Produto', 'Estoque', 'Preço original',	'Preço promocional variações', 'Ações']}
  >
    <TableGroup>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
    </TableGroup>
    <TableGroup>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
    </TableGroup>
    <TableGroup>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
    </TableGroup>
    <TableGroup>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
      <BodyColumn>content</BodyColumn>
    </TableGroup>
  </Table>
</Playground>