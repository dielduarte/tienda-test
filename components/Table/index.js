import Table, { TableGroup, BodyColumn, TableContent, TableImg } from './Table';

export { TableGroup, BodyColumn, TableContent, TableImg };
export default Table;
