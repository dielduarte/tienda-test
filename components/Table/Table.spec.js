import { shallow } from 'enzyme';
import 'jest-styled-components';

import Table, { TableGroup, BodyColumn } from './Table';

describe('Table', () => {
  it('shoud render correctly', () => {
    const wrapper = shallow(
      <Table
        headers={[
          'Produto',
          'Estoque',
          'Preço original',
          'Preço promocional variações',
          'Ações'
        ]}
      >
        <TableGroup>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
        </TableGroup>
        <TableGroup>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
          <BodyColumn>content</BodyColumn>
        </TableGroup>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
