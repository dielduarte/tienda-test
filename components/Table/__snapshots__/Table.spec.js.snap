// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table shoud render correctly 1`] = `
<styled.table>
  <thead>
    <styled.tr>
      <styled.th
        key="0"
      >
        Produto
      </styled.th>
      <styled.th
        key="1"
      >
        Estoque
      </styled.th>
      <styled.th
        key="2"
      >
        Preço original
      </styled.th>
      <styled.th
        key="3"
      >
        Preço promocional variaçoões
      </styled.th>
      <styled.th
        key="4"
      >
        Ações
      </styled.th>
    </styled.tr>
  </thead>
  <tbody>
    <styled.tr>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
    </styled.tr>
    <styled.tr>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
      <styled.td>
        content
      </styled.td>
    </styled.tr>
  </tbody>
</styled.table>
`;
